### DISCUSSION TITLE
Cannot set CheckBox path color.

### DISCUSSION THREAD

**Discussion by Nickelony** [AvaloniaUI/Avalonia#8168] [Q&A]
Basically I use this code:
```xaml
<Style x:Class="DarkUIColorScheme">
	<Style.Resources>
		<SolidColorBrush x:Key="GreyHighlight">#7A8084</SolidColorBrush>
	</Style.Resources>
</Style>
```
```xaml
<Style Selector="CheckBox:checked /template/ Path#checkMark">
    <Setter Property="Fill" Value="{DynamicResource GreyHighlight}" />
</Style>
<Style Selector="CheckBox:indeterminate /template/ Rectangle#indeterminateMark">
    <Setter Property="Fill" Value="{DynamicResource GreyHighlight}" />
</Style>
```
which should change the color of the check mark to a grey... but it doesn't:
![image](https://user-images.githubusercontent.com/20436882/169656795-184aa1da-c88f-44a0-a239-d06587e0d883.png)

Changing any other property works perfectly fine, but not the Fill color. What should I do? Is this a bug?
I'm using Default Theme.

**Comment by maxkatz6** [comment]
Hi!
Some details of default templates use locally set values for the properties, which isn't easy to redefine with a style. Especially if it's an old default theme. For this theme it might be easier to redefine only resource (HighlightBrush) with your value or replace whole template.

More information is in the docs:
https://docs.avaloniaui.net/docs/styling/troubleshooting#locally-set-properties-override-styles
And this specific local value:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Default/Controls/CheckBox.xaml#L23

**Reply by maxkatz6** [reply]
This restriction will be relaxed with this PR https://github.com/AvaloniaUI/Avalonia/pull/7679 , but it will introduce quite a lot of breaking changes, so no guarantee on it for now.
